src/utils/readFile.ts

```diff
@@ -0,0 +1,35 @@
+import * as RNFS from 'react-native-fs'
+import Logger from 'src/utils/Logger'
+
+/**
+ * Reads and returns a portion of the input file.
+ */
+const readChunk = async (file: string, length: number, position: number) => {
+  try {
+    return await RNFS.read(file, length, position, 'utf8')
+  } catch (error) {
+    Logger.error('Logger@readChunk', 'readChunk error', error as Error)
+    return ''
+  }
+}
+
+/**
+ * Reads the logs in chunks to a string and returns the string.
+ * [Credits]{@link: https://stackoverflow.com/a/56284734}
+ */
+export const readFileChunked = async (rnLogsSrc = Logger.getReactNativeLogFilePath()) => {
+  try {
+    const length = 4096
+    let chunk = ''
+    let fileContents = ''
+    if (await RNFS.exists(rnLogsSrc)) {
+      do {
+        chunk = await readChunk(rnLogsSrc, length, fileContents.length)
+        fileContents += chunk
+      } while (chunk.length > 0)
+    }
+    return fileContents
+  } catch (error) {
```

Comment:
why swallow the error instead of bubbling it up?

Reply:
Fixed in [82a3162](https://github.com/valora-inc/wallet/pull/2680/commits/82a3162bcb042760961e008c0132769d04f6ba48)!